redbeams/src/main/java/com/sequenceiq/redbeams/service/dbserverconfig/DatabaseServerConfigService.java

```diff
@@ -220,8 +220,16 @@ public DatabaseServerConfig getByCrn(String resourceCrn) {
         return repository.findByResourceCrn(resourceCrn);
     }
 
-    public Optional<DatabaseServerConfig> getByClusterCrn(String clusterCrn) {
-        return repository.findByClusterCrn(clusterCrn);
+    public DatabaseServerConfig getByClusterCrn(String environmentCrn, String clusterCrn) {
+        DatabaseServerConfig databaseServerConfig = getByClusterCrnOptional(environmentCrn, clusterCrn)
+                .orElseThrow(() -> new NotFoundException(String.format("No %s found with cluster CRN '%s' in environment '%s'",
+                        DatabaseServerConfig.class.getSimpleName(), clusterCrn, environmentCrn)));
+        MDCBuilder.buildMdcContext(databaseServerConfig);
+        return databaseServerConfig;
+    }
+
+    public Optional<DatabaseServerConfig> getByClusterCrnOptional(String environmentCrn, String clusterCrn) {
```

Comment:
it would be nice to follow the repository's pattern here: `find` returns `Optional`, while `get` returns the pojo and fails in case there is none. so the method name would look like here: `findByEnvironmentIdAndClusterCrn`, same as the repo

Reply:
findByEnvironmentCrnAndClusterCrn right ?

Reply:
right, I copied the repository method name. unfortunately in redbeams it called `environmentId` instead of crn...

Reply:
I see you @lacikaaa . I see you ...